pkg/controller/endpointslice/reconciler.go

```diff
@@ -120,13 +120,19 @@ func (r *reconciler) reconcileByAddressType(service *corev1.Service, pods []*cor
 	slicesToDelete := []*discovery.EndpointSlice{}
 
 	// Build data structures for existing state.
+	currentAnnotations := map[string]string{}
 	existingSlicesByPortMap := map[endpointutil.PortMapKey][]*discovery.EndpointSlice{}
 	numExistingEndpoints := 0
 	for _, existingSlice := range existingSlices {
 		if ownedBy(existingSlice, service) {
 			epHash := endpointutil.NewPortMapKey(existingSlice.Ports)
 			existingSlicesByPortMap[epHash] = append(existingSlicesByPortMap[epHash], existingSlice)
 			numExistingEndpoints += len(existingSlice.Endpoints)
+			// Get current annotations, in case that there are multiple slices is
+			// the user responsability to keep them consistent.
+			if len(existingSlice.Annotations) > 0 {
+				currentAnnotations = existingSlice.Annotations
+			}
```

Comment:
This feels like it's going to result in lots of inconsistent behavior that is difficult to troubleshoot. This appears to assign the non-empty annotations of the last slice in this list. I'm primarily concerned about the likelihood of inconsistent annotations across slices. It would be very easy to get into a state where each slice has a different number of annotations. This would all be based on the rather arbitrary logic of which slice was last in the list. Annotations on existing slices would also never be updated by this controller - further ensuring inconsistency. 

I'm used to annotations being used to indicate unique characteristics of a specific resource. Replicating annotations from Service down to EndpointSlices also seems like it could be reasonable since there is clear hierarchy. Unfortunately trying to keep annotations in sync across EndpointSlices feels impossible to do safely and reliably.